openquake/server/views.py

```diff
@@ -145,6 +148,52 @@ def get_engine_version(request):
     return HttpResponse(oqversion)
 
 
+@csrf_exempt
+@cross_domain_ajax
+@require_http_methods(['POST'])
+def validate_nrml(request):
+    """
+    Leverage oq-risklib to check if a given xml text is a valid nrml
+
+    :param request:
+        a `django.http.HttpRequest` object containing the mandatory
+        parameter 'xml_text': the text of the xml to be validated as nrml
+
+    :returns: a JSON object, containing:
+        * 'valid': a boolean indicating if the provided text is a valid nrml
+        * 'error_msg': the validation error message, if any error was found
+                       (None otherwise)
+        * 'error_line': line of the given xml where the error was found
+                        (None if no error was found)
+    """
+    xml_text = request.POST.get('xml_text')
+    if not xml_text:
+        return HttpResponseBadRequest(
+            'Please provide the "xml_text" parameter')
+    xml_file = writetmp(xml_text, suffix='.xml')
+    response_data = {}
+    try:
+        nrml.read(xml_file)
+    except Exception as exc:
+        # get the exception message
+        exc_msg = exc.args[0]
+        # if it is not a simple string, let's assume it's UTF8 eoncoded
+        if not isinstance(exc_msg, bytes):
+            exc_msg = exc_msg.encode('utf8')
+        error_msg = exc_msg.split(', line')[0]
+        error_line = int(
+            re.search(r'line \d+', exc_msg).group(0).split()[1])
+        response_data['error_msg'] = error_msg
```

Comment:
I would be more careful here. If the error message does not contain a line number, return '-1' as error_line.